cdc/generation.cc

```diff
@@ -774,18 +774,18 @@ future<> generation_service::after_join(std::optional<cdc::generation_id>&& star
     _cdc_streams_rewrite_complete = maybe_rewrite_streams_descriptions();
 }
 
-future<> generation_service::on_join(gms::inet_address ep, gms::endpoint_state ep_state) {
+future<> generation_service::on_join(gms::inet_address ep, gms::endpoint_state ep_state, gms::permit_id pid) {
     assert_shard_zero(__PRETTY_FUNCTION__);
 
     auto val = ep_state.get_application_state_ptr(gms::application_state::CDC_GENERATION_ID);
     if (!val) {
         return make_ready_future();
     }
 
-    return on_change(ep, gms::application_state::CDC_GENERATION_ID, *val);
+    return on_change(ep, gms::application_state::CDC_GENERATION_ID, *val, pid);
 }
 
-future<> generation_service::on_change(gms::inet_address ep, gms::application_state app_state, const gms::versioned_value& v) {
+future<> generation_service::on_change(gms::inet_address ep, gms::application_state app_state, const gms::versioned_value& v, gms::permit_id) {
```

Comment:
@bhalevy Please document how the listeners are supposed to do with the permit_id?

Reply:
okay (in a follow-up)

Reply:
I already documented this: https://github.com/scylladb/scylladb/blob/39ca07c49b257d37856164ea5b845f7b98224b98/gms/i_endpoint_state_change_subscriber.hh#L28-L33
Is it lacking details, or do you think it's also needed elsewhere (but I rather than have one authoritative source of information than spread it around)

Reply:
OK. I think this is good enough.